examples/step-74/step-laplace-FEIV.cc

```diff
@@ -0,0 +1,1245 @@
+              }             
+          }
+        }
+                                       
+
+    private:
+    Functions::LSingularityFunction ref;
+    Test_Case test_case;
+                   
+  };
+
+
+ /* 
+  template <int dim>
+  class Beta
+  {
+  public:
+    Beta () {}
+    void value_list (const std::vector<Point<dim> > &points,
+                     std::vector<Point<dim> > &values) const;
+  };
+
+  template <int dim>
+  void Beta<dim>::value_list(const std::vector<Point<dim> > &points,
+                             std::vector<Point<dim> > &values) const
+  {
+    Assert(values.size()==points.size(),
+           ExcDimensionMismatch(values.size(),points.size()));
+
+    for (unsigned int i=0; i<points.size(); ++i)
+      {
+        const Point<dim> &p=points[i];
+        Point<dim> &beta=values[i];
+
+        beta(0) = -p(1);
+        beta(1) = p(0);
+        beta /= std::sqrt(beta.square());
+      }
+  }
+*/
+  // @sect3{Equation data}
+  //
+  // First, we define a class describing the inhomogeneous boundary
+  // data. Since only its values are used, we implement value_list(), but
+  // leave all other functions of Function undefined.
+  // template <int dim>
+  // class BoundaryValues:  public Function<dim>
+  // {
+  // public:
+  //   BoundaryValues () {};
+  //   virtual void value_list (const std::vector<Point<dim> > &points,
+  //                            std::vector<double> &values,
+  //                            const unsigned int component=0) const;
+  // };
+
+  // Given the flow direction, the inflow boundary of the unit square
+  // $[0,1]^2$ are the right and the lower boundaries. We prescribe
+  // discontinuous boundary values 1 and 0 on the x-axis and value 0 on the
+  // right boundary. The values of this function on the outflow boundaries
+  // will not be used within the DG scheme.
+  // template <int dim>
+  // void BoundaryValues<dim>::value_list(const std::vector<Point<dim> > &points,
+  //                                      std::vector<double> &values,
+  //                                      const unsigned int) const
+  // {
+  //   Assert(values.size()==points.size(),
+  //          ExcDimensionMismatch(values.size(),points.size()));
+
+  //   for (unsigned int i=0; i<values.size(); ++i)
+  //     {
+  //         values[i] = 0; //sin(2*numbers::PI*points[i](0));
+
+  //     }
+  // }
+  // @sect3{The AdvectionProblem class}
+  //
+  // After this preparations, we proceed with the main class of this program,
+  // called AdvectionProblem. It is basically the main class of step-6. We do
+  // not have a ConstraintMatrix, because there are no hanging node
+  // constraints in DG discretizations.
+
+  // Major differences will only come up in the implementation of the assemble
+  // functions, since here, we not only need to cover the flux integrals over
+  // faces, we also use the MeshWorker interface to simplify the loops
+  // involved.
+template <int dim>
+struct ScratchData1
```

Comment:
not sure if this is used, but we should use the ``dealii:MeshWorker::ScratchData<dim, spacedim>``